### PR TITLE
Jd 2 sidebar search

### LINKED ISSUE #2: Search Dropdown
**Is your feature request related to a problem? Please describe.**
This feature will enable pokemon search via dropdown menu.

### Steps to complete
- Enable sorting the dropdown list
- [Install Typeahead ](https://ericgio.github.io/react-bootstrap-typeahead/)
- Render Typeahead in the markup
- Render pokemon selected in the sidebar

### PR DESCRIPTION
**Pull request by JayDampitan**
## Changes
1. Added useSelector to retrieve all pokemon from redux
2. Added Typeahead extension 
3. Added a function to sort all pokemon alphabetically.

## Purpose
Users should be able to search pokemon using the side bar input using typeahead and render selected pokemon to the Sidebar component. 

## Approach
Used Redux useSelector to retrieve all pokemon and a function to sort them out alphabetically. Adding Typeahead from bootstrap to render pokemon on the Sidebar. 

Closes #2

### CODE REVIEW

File: pokeapp/package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "node-sass": "^6.0.1",
     "pokeapi-js-wrapper": "^1.2.0",
     "react": "^17.0.2",
+    "react-bootstrap-typeahead": "^5.2.0",
```
Comment (BlackCubes) [inline pokeapp/package.json:13]: Bootstrap 👀

File: pokeapp/src/components/Sidebar.js [hunk 2]
```diff
@@ -1,22 +1,67 @@
...
+import { Typeahead } from "react-bootstrap-typeahead";
+import "react-bootstrap-typeahead/css/Typeahead.css";
+
+const sortAtoZ = (list) => {
+  const sortedAtoZ = list.sort((name1, name2) => {
+    if (name1 < name2) return -1;
+    if (name1 > name2) return 1;
+    return 0;
+  });
+  return sortedAtoZ;
```
Comment (Lap343) [inline pokeapp/src/components/Sidebar.js:20]: nice sorting method

**Review by Lap343 — APPROVED** [review]
Good sidebar search but also good code on sorting the search

File: pokeapp/src/components/Sidebar.js [hunk 3]
```diff
@@ -1,22 +1,67 @@
...
+    });
+    setPokeNameList(sortAtoZ(names));
+  };
+
+  // This event gets triggered by pressing the enter key
+  const enterSubmit = (e) => {
+    if (e.key === "Enter") {
+      updateSidebarPokemon(allPokemon, pokeSearch);
+    }
+  };
```
Comment (BlackCubes) [inline pokeapp/src/components/Sidebar.js:55]: Interesting

File: pokeapp/src/components/Sidebar.js [hunk 4]
```diff
@@ -1,22 +1,67 @@
...
   makeUpperCase,
   isMoreInfo,
   setIsMoreInfo,
+  setSelectedPokemon,
 }) => {
+  const [showPokemon, setShowPokemon] = useState(false);
+  const [pokeNameList, setPokeNameList] = useState([]);
+
+  // Endpoint. This is the string that will be used by the type ahead
+  const [pokeSearch, setPokeSearch] = useState([]);
```
Comment (kmausolf) [inline pokeapp/src/components/Sidebar.js:34]: Good job recognizing that this needs documentation, as its context is not immediately apparent.

**Review by BlackCubes — APPROVED** [review]
This whole app looks narly. Can't wait to see it!

File: pokeapp/src/components/Sidebar.js (hunk 3 above)
Comment (kmausolf) [inline pokeapp/src/components/Sidebar.js:55]: Very nice, this is a feature that many developers leave out that many users expect!

File: pokeapp/src/components/Sidebar.js (hunk 2 above)
Comment (JorgeLVilla) [reply pokeapp/src/components/Sidebar.js:20]: I have to agree with team captain. Clean!

**Review by JorgeLVilla — APPROVED** [review]
Great job as always Jay!

File: pokeapp/src/components/Sidebar.js [hunk 5]
```diff
@@ -1,22 +1,67 @@
...
+  const enterSubmit = (e) => {
+    if (e.key === "Enter") {
+      updateSidebarPokemon(allPokemon, pokeSearch);
+    }
+  };
+
+  const updateSidebarPokemon = (pokemonList, searchValue) => {
+    for (const pokemon of pokemonList) {
+      if (searchValue[0] === pokemon.name) {
+        setSelectedPokemon(pokemon);
```
Comment (kmausolf) [inline pokeapp/src/components/Sidebar.js:62]: Let's add a `break;` statement after line 60.  We should do this to stop the for loop from running after discovering the first occurrence of the search string.

Comment (kmausolf) [reply pokeapp/src/components/Sidebar.js:62]: If we don't it'll potentially call `setSelectedPokemon` for each matching string (if there are any), or it'll keep searching even after it's found a match.

File: pokeapp/src/components/Sidebar.js [hunk 6]
```diff
@@ -25,8 +72,39 @@ const SideBar = ({
...
+              className="sidebar-dropdown-new"
+              onKeyPress={(e) => {
+                enterSubmit(e);
+              }}
+            >
+              <Typeahead
+                id="basic-typeahead-single"
+                labelKey="name"
+                options={pokeNameList}
+                placeholder={selectedPokemon.name}
```
Comment (kmausolf) [inline pokeapp/src/components/Sidebar.js:92]: If there's no `selectedPokemon`, this may throw an error.  I'd set it to a static string like "Search..."

**Review by BlackCubes — COMMENTED** [review]
Nice update!

**Review by Gjhnsn — COMMENTED** [review]
Looking good, great work!

**Pull request merged**